docs: proofread market user guides (#71)
NilFoundation/proof-market-toolchain: gitbook-docs/market/user-guides/circuit-developer.md

# Circuit developer

This is an entity that prepares circuits for Proof Market.
The most efficient way to create a circuit definition for Proof Market is
[zkLLVM](https://github.com/NilFoundation/zkllvm) circuit compiler.
You can find the required dependencies and build instructions in the
[zkLLVM's repository](https://github.com/NilFoundation/zkllvm).

Anyone can generate circuits.
They are serialized & published on Proof Market, allowing reuse of the circuits by
all proof requesters.

> ## Create a new circuit using [zkLLVM](https://github.com/NilFoundation/zkllvm)

Reviewer: It's hard to see a link in the heading.
```suggestion
## Create a circuit

Write a circuit and compile it with [zkLLVM](https://github.com/NilFoundation/zkllvm).
For more information on circuit development, read the [zkLLVM documentation](
https://docs.nil.foundation/zkllvm/circuit-development/circuit-generation).

```

Author: Fixed

```bash
make -C ${ZKLLVM_BUILD:-build} <circuit target name> -j$(nproc)
```

## Prepare a statement with a circuit description for Proof Market

> Circuits are stored as a _statement_ structure on Proof Market.
> Example of a statement description can be found in the `/example` directory.

Reviewer: We're still struggling with terminology.

What's a statement structure? What's a statement description?

I understand that a statement is a data structure internally, but it should not bother circuit developers. For them, it's just an artifact that is produced from a circuit and accepted by Proof Market.

By its meaning and purpose, a statement is a description of a circuit combined with the circuit itself. But in the `./example` directory we see the statements, not their descriptions.

So, let's reduce the number of used terms. Just statements. 

```suggestion
Circuits are stored on Proof Market in the form of statements.
Example statements can be found in the `./example` directory.
```

Oh, and paths like `/path` are paths from the root of filesystem or a website. Paths from the project root or current directory are `./path/`, `./path` or `path/`. Just `path.txt` can be okay for files, but not so much for directories.

Author: Fixed

Author: About paths - we should probably check and fix other PMT's docs

```bash

> python3 scripts/prepare_statement.py -c <zkllvm output> \

Reviewer: let's wrap the first line as well

Author: Fixed

    -o <statement description file> \ 

>     -n <statement name> -t <circuit type>

Reviewer: ```suggestion
    -n <statement name> -t <circuit type> \
    --private | --public
```

Author: Thanks, fixed

```

> Make sure to provide the necessary information listed in the output statement file.

Reviewer: I don't understand what I should provide. What's necessary?

Author: I wish I knew :)

I think this sentence is about circuit's parameters which are actually the statement_tools' parameters:
    --circuit <zkllvm output> \
    --type <circuit type> \
    --output <statement description file> \
    --name <statement name> \
    --private | --public

So I guess this sentence is redundant here?

> ## Publish a statement to Proof Market

Reviewer: I have a feeling that we publish statements _on_ Proof Market. Like this https://www.collinsdictionary.com/dictionary/english/put-on-the-market

Author: Thanks, fixed

{% hint style="info" %}

> Make sure you've performed the [authentication setup](sign-up.md) before continuing.

Reviewer: This could be said simpler. Like,

- [setup the authentication]() before continuing
- Sign up and keep the authentication files to use command line tools.
— You must sign up and keep the authentication files to use command line tools.

Author: Since it's not all CLI tools that require authentication, I updated this warnings as follows:

```
[Sign up](sign-up.md) and keep the authentication files in order to use
the following command line tools.
```

{% endhint %}

> You can find an authorization file example in the `/example` directory.

Reviewer: authorization or authentication?

Author: Authentication, of course. It's a json with login and pass https://github.com/NilFoundation/proof-market-toolchain/blob/master/example/auth.json

Author: Fixed

> The statement can be pushed to Proof Market via the Python script `statement_tools.py`:

Reviewer: I wanted to say, “use post or publish instead of push”, but the command is literally `push`.

Author: Yep, but I think publish will be better here

Author: Updated

```bash
python3 scripts/statement_tools.py push 
    --file <json file with statement description> 
```

> The script will return an object containing a _\_key_ field — a unique descriptor of the statement.

Reviewer: And I have a feeling that it can be said even simpler.

> The script will return an object containing a _\_key_ field — a unique descriptor of the statement.

Reviewer: Oh, and I didn't make that change yet, so for now the script doesn't return the id. It just prints
```
Statement from /opt/zkllvm-template/build/template.json was pushed.
```

> The script will return an object containing a _\_key_ field — a unique descriptor of the statement.

Author: Ok, updated

## Retrieving statements

A list of all available statements can be obtained by this command:

```bash
python3 scripts/statement_tools.py get
```

To retrieve a selected statement/circuit definition and its metadata
uploaded upon the statement's publishing:

> ```bash
> python3 scripts/statement_tools.py get \
>     --key <key of the statement> \
>     -o <output file> 
> ```

Reviewer: this one is almost perfect! There's one little detail to make the example easier to understand:

```suggestion
```bash
python3 scripts/statement_tools.py get \
    --key <key of the statement> \
    --output <output file> 
```
```

Author: Fixed
